modules/alloy/src/main/ts/ephox/alloy/positioning/mode/SelectionAnchor.ts

```diff
@@ -32,20 +39,41 @@ const placement = (component: AlloyComponent, anchorInfo: SelectionAnchor, origi
 
   const selectionBox = getAnchorSelection(win, anchorInfo).bind((sel) => {
     // This represents the *visual* rectangle of the selection.
-    const optRect = WindowSelection.getBounds(win, SimSelection.exactFromRange(sel)).orThunk(() => {
-      const x = SugarElement.fromText(Unicode.zeroWidth);
-      Insert.before(sel.start, x);
-      // Certain things like <p><br/></p> with (p, 0) or <br>) as collapsed selection do not return a client rectangle
-      const rect = WindowSelection.getFirstRect(win, SimSelection.exact(x, 0, x, 1));
-      Remove.remove(x);
-      return rect;
-    });
+    if (isSimRange(sel)) {
+      const optRect = WindowSelection.getBounds(win, SimSelection.exactFromRange(sel)).orThunk(() => {
+        const zeroWidth = SugarElement.fromText(Unicode.zeroWidth);
+        Insert.before(sel.start, zeroWidth);
+        // Certain things like <p><br/></p> with (p, 0) or <br>) as collapsed selection do not return a client rectangle
+        const rect = WindowSelection.getFirstRect(win, SimSelection.exact(zeroWidth, 0, zeroWidth, 1));
+        Remove.remove(zeroWidth);
+        return rect;
+      });
+
+      return optRect.bind((rawRect) => {
+        return ContentAnchorCommon.getBox(rawRect.left, rawRect.top, rawRect.width, rawRect.height);
+      });
+    } else {
+      const selectionRect = Obj.map(sel, (cell) => cell.dom.getBoundingClientRect());
 
-    return optRect.bind((rawRect) => ContentAnchorCommon.getBox(rawRect.left, rawRect.top, rawRect.width, rawRect.height));
+      const bounds = {
+        left: Math.min(selectionRect.firstCell.left, selectionRect.lastCell.left),
+        right: Math.max(selectionRect.firstCell.right, selectionRect.lastCell.right),
+        top: Math.min(selectionRect.firstCell.top, selectionRect.lastCell.top),
+        bottom: Math.max(selectionRect.firstCell.bottom, selectionRect.lastCell.bottom)
+      };
+
+      return ContentAnchorCommon.getBox(bounds.left, bounds.top, bounds.right - bounds.left, bounds.bottom - bounds.top);
+    }
   });
 
   const targetElement = getAnchorSelection(win, anchorInfo)
-    .bind((sel) => SugarNode.isElement(sel.start) ? Optional.some(sel.start) : Traverse.parentElement(sel.start));
+    .bind((sel) => {
+      if (isSimRange(sel)) {
+        return SugarNode.isElement(sel.start) ? Optional.some(sel.start) : Traverse.parentElement(sel.start);
+      }
+
+      return Optional.some(sel.firstCell);
```

Comment:
Probably best for consistency

```suggestion
      } else {
            return Optional.some(sel.firstCell);
      }
```